exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscloudwatchlogsexporter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type exporter struct {
+	config *Config
+	logger *zap.Logger
+
+	startOnce sync.Once
+	client    *cloudwatchlogs.CloudWatchLogs // available after startOnce
+
+	seqTokenMu sync.Mutex
+	seqToken   string
+}
+
+func (e *exporter) Start(ctx context.Context, host component.Host) error {
+	var startErr error
+	e.startOnce.Do(func() {
+		awsConfig := &aws.Config{}
+		if e.config.Region != "" {
+			awsConfig.Region = aws.String(e.config.Region)
+		}
+		if e.config.Endpoint != "" {
+			awsConfig.Endpoint = aws.String(e.config.Endpoint)
+		}
+		awsConfig.MaxRetries = aws.Int(1) // retry will be handled by the collector queue
+		sess, err := session.NewSession(awsConfig)
+		if err != nil {
+			startErr = err
+			return
+		}
+		e.client = cloudwatchlogs.New(sess)
+
+		e.logger.Debug("Retrieving Cloud Watch sequence token")
+		out, err := e.client.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+			LogGroupName:        aws.String(e.config.LogGroupName),
+			LogStreamNamePrefix: aws.String(e.config.LogStreamName),
+		})
+		if err != nil {
+			startErr = err
+			return
+		}
+		if len(out.LogStreams) == 0 {
+			startErr = errors.New("cannot find log group and stream")
+			return
+		}
+		stream := out.LogStreams[0]
+		e.seqToken = *stream.UploadSequenceToken
+	})
+	return startErr
+}
+
+func (e *exporter) Shutdown(ctx context.Context) error {
+	// TODO(jbd): Signal shutdown to flush the logs.
+	return nil
+}
+
+func (e *exporter) PushLogs(ctx context.Context, ld pdata.Logs) (droppedTimeSeries int, err error) {
+	// TODO(jbd): Relax this once CW Logs support ingest
+	// without sequence tokens.
+	e.seqTokenMu.Lock()
+	defer e.seqTokenMu.Unlock()
+
+	logEvents, dropped := logsToCWLogs(e.logger, ld)
+	if len(logEvents) == 0 {
+		return 0, nil
+	}
+	if dropped > 0 {
+		e.logger.Warn("CloudWatch Logs exporter dropped log records", zap.Any("count", dropped))
```

Comment:
Please change to Debug or return an error to indicate a drop. We don't want to flood the logs if there are destination errors. Alternatively create a rate-limited logger if want it to be Warn.

Reply:
Removing this.